src/pages/SettingsPage/UsersSettings/UserList.jsx

```diff
@@ -29,10 +29,21 @@ const UserList = ({
 }) => {
   // Selection
   const selection = useMemo(
-    () => userList.filter((user) => selectedUsers.includes(user.name)),
+    () => {
+      return userList.filter((user) => selectedUsers.includes(user.name))
+    },
     [selectedUsers, selectedProjects, userList],
   )
 
+  const onContextMenu = (e) => {
+    if (selectedUsers.includes(e.data.name)) {
+      onSelectUsers([e.data.name, ...selectedUsers])
```

Comment:
This if statement checks if the right click is within the selection (you are right clicking something already selected). If this is true then we want to do nothing because the selection should not change. This code duplicates the selection state with the same users over and over again on evert right click, look at the console that is logging `selectedUsers`.


https://github.com/user-attachments/assets/10c56248-1cd0-420e-89a6-a70726f05e12